src/shared/functions.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python3
+
+"""
+Takes a releases trigger file and created a mapping of all the
+OCI tags that are marked to be release.
+"""
+
+import json
+import os
+import re
+import subprocess
+from collections import defaultdict
+import yaml
+from src.image.utils.schema.triggers import ImageSchema, KNOWN_RISKS_ORDERED
+
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+
+
+class BadChannel(Exception):
+    """Error validating release channel."""
+
+
+def get_all_revision_tags(file_all_revision_tags):
+    """
+    This function permits to get all the revision tags.
+    """
+    with open(file_all_revision_tags, encoding="UTF-8") as rev_tags_f:
+        return rev_tags_f.read().strip().rstrip(",").lstrip(",").split(",")
+
+
+def get_all_releases(file_all_releases):
+    """
+    This function permits to get all the releases.
+    """
+    try:
+        with open(file_all_releases, encoding="UTF-8") as all_releases_fd:
+            return json.load(all_releases_fd)
+    except FileNotFoundError:
+        return {}
+
+
+def get_image_trigger(file_image_trigger):
+    """
+    This function permits to parse and validate the image_trigger file.
+    """
+    with open(file_image_trigger, encoding="UTF-8") as trigger:
+        return ImageSchema(**yaml.safe_load(trigger))
+
+
+def get_tag_mapping_from_all_releases(all_releases_dict):
+    """
+    This function permits to map the tag from all_releases
+    """
+    mapping_from_all_releases = {}
+    for track, risks in all_releases_dict.items():
+        for risk, values in risks.items():
+            if risk in KNOWN_RISKS_ORDERED:
+                tag = f"{track}_{risk}"
+
+                mapping_from_all_releases[tag] = values["target"]
+    return mapping_from_all_releases
+
+
+def get_revision_to_track(all_revision):
+    """
+    This function permits the conversion of the revision to track.
+    """
+    revision_track = {}
+    for track_revision in all_revision:
```

Comment:
fix variable name `all_revision`. should be plural, and should allude to what it actually is - and that's not a list of revision numbers

Reply:
yeah, true i have change to all_revisions_tags